File: Programas/NoInformados/Rompecabezas/1. Busqueda en Profundidad v1.0.0/C++ DFS/Principal.cpp
#include "./Header/Rompecabezas.h"
#include "./Header/DFS.h"

int main(){

    Rompecabezas * n1 = new Rompecabezas( nullptr , ' ' , 0 );
    n1->revolverCeldas(3);

    std::cout << "Nodo Inicial" << "\n";
    n1->mostrarRompecabezas();

    std::cout << "Ruta" << "\n";

    std::vector<Rompecabezas*> ruta = DFS::Bucar( n1 , new Rompecabezas( nullptr , ' ' , 0 ) );

    for( int i = ruta.size()-1 ; i >= 0 ; i-- ){
        ruta[i]->mostrarRompecabezas();
    }

}

File: Programas/NoInformados/Rompecabezas/1. Busqueda en Profundidad v1.0.0/JAVA DFS/Principal.java
import java.util.ArrayList;

public class Principal {
    public static void main(String[] args) {
        
        Rompecabezas nodoInicial = new Rompecabezas( null , null , 0 );
        nodoInicial.revolverCeldas(3);
        
        System.out.println( "Nodo Inicial: " );
        nodoInicial.mostrarRompecabezas();

        ArrayList<Rompecabezas> camino = DFS.Bucar( nodoInicial , new Rompecabezas( null , null , 0 ) );

        System.out.println( "Camino:" );

        for( int i = 0 ; i < camino.size() ; i++ ){
            camino.get(i).mostrarRompecabezas();
        }

    }
}
